### PR TITLE
refactor: use cypress design button in launchpad

### PR DESCRIPTION
**Pull request by elevatebart**
Related to https://github.com/cypress-io/cypress-design/issues/189

### Assumptions

- `pending` is always `disabled` so the variants are merged
- size `sm` is 24 px tall, set the size accordingly
- size `lg` is 40 px tall, set the size accordingly
- the default size is different so I added size=32 when there is none
- instead of using prefix and suffix icons, I use declarative html, the way it is described in [the documentation](https://design.cypress.io/components/vue/Button.html#usage)
- In global mode I replaced the "gray" (disabled) button with a `light-indigo`
- Instead of embedding behavior for external links inside the "Button", I use the `useExternalLink()` hook directly when necessary.

### CODE REVIEW

File: packages/launchpad/src/components/code/FileRow.vue [hunk 1]
```diff
@@ -98,6 +100,9 @@ const props = defineProps<{
   fileExtension: string
 }>()
 
+const LEARN_MORE_URL = 'https://on.cypress.io/guides/configuration'
```
Comment (emilyrohrbough) [inline packages/launchpad/src/components/code/FileRow.vue]: nit: define this constant at the top of the file under the imports

File: packages/launchpad/src/setup/ButtonBar.vue [hunk 2]
```diff
@@ -76,7 +73,7 @@ const props = withDefaults(defineProps<{
   alt?: string
   altFn?: (value: boolean) => void
   canNavigateForward?: boolean
-  mainVariant?: ButtonVariants
+  mainVariant?: ButtonVariants | 'pending'
```
Comment (emilyrohrbough) [inline packages/launchpad/src/setup/ButtonBar.vue:73]:
```suggestion
  mainVariant?: ButtonVariants
```

File: packages/launchpad/src/setup/ButtonBar.vue [hunk 3]
```diff
@@ -62,10 +57,12 @@
 
 <script lang="ts" setup>
 import { ref } from 'vue'
-import type { ButtonVariants } from '@cy/components/Button.vue'
-import Button from '@cy/components/Button.vue'
+import type { VariantClassesTable } from '@cypress-design/vue-button'
```
Comment (emilyrohrbough) [inline packages/launchpad/src/setup/ButtonBar.vue]:
```suggestion
import type { VariantClassesTable as ButtonVariants } from '@cypress-design/vue-button'
```

File: packages/launchpad/src/setup/OpenBrowserList.vue [hunk 4]
```diff
@@ -154,7 +153,7 @@
 
 <script lang="ts" setup>
 import { useI18n } from '@cy/i18n'
-import Button from '@packages/frontend-shared/src/components/Button.vue'
```
Comment (emilyrohrbough) [inline packages/launchpad/src/setup/OpenBrowserList.vue:157]: Is this button still used or can we delete this code? `'@packages/frontend-shared/src/components/Button.vue'`

File: yarn.lock [hunk 5]
```diff
@@ -2306,7 +2306,7 @@
   dependencies:
     "@cypress-design/css" "^0.15.1"
 
-"@cypress-design/vue-button@^0.10.2":
+"@cypress-design/vue-button@0.10.2", "@cypress-design/vue-button@^0.10.2":
```
Comment (emilyrohrbough) [inline yarn.lock]: Can we use an open range in the package.json so this matches?

Comment (elevatebart) [reply yarn.lock]: fixed by 3c7bd26

File: packages/launchpad/src/setup/OpenBrowserList.vue (hunk 4 above)
Comment (elevatebart) [reply packages/launchpad/src/setup/OpenBrowserList.vue:157]: To keep the PR of a decent size, I only worked on the LaunchPad for now. 
In a 2nd PR, I will process the app and delete the unused Button.

File: packages/launchpad/src/components/code/FileRow.vue (hunk 1 above)
Comment (elevatebart) [reply packages/launchpad/src/components/code/FileRow.vue]: fixed in a4500825116f1c10fc5814711ff31b4769d9173e

**Review by emilyrohrbough — APPROVED** [review]

**Review by ryanjwilke — APPROVED** [review]
Code looks good to me. I tried looking at Percy diffs but it appears that the build is stuck possibly. Feel free to ping me if you have a link to Percy diffs once those are ready. If there aren't really any visual diffs worth noting, then I'm good 👍

**Pull request merged**